modules/tiles/test/tileset/traversers/tileset-traverser.spec.js

```diff
@@ -0,0 +1,49 @@
+import test from 'tape-promise/tape';
+import {WebMercatorViewport} from '@deck.gl/core';
+import {load} from '@loaders.gl/core';
+import {Tiles3DLoader} from '@loaders.gl/3d-tiles';
+import {Tileset3D} from '@loaders.gl/tiles';
+
+import TilesetTraverser from '../../../src/tileset/traversers/tileset-traverser';
+import {getFrameState} from '../../../src/tileset/helpers/frame-state';
+
+// Parent tile with content and four child tiles with content
+const TILESET_URL = '@loaders.gl/3d-tiles/test/data/Tilesets/Tileset/tileset.json';
+
+test('Tileset3D#traverser base class', async (t) => {
+  const tilesetJson = await load(TILESET_URL, Tiles3DLoader);
+
+  // Create Tileset3D to have initialized Tile3Ds tree
+  const tileset = new Tileset3D(tilesetJson);
+
+  const traverser = new TilesetTraverser({
+    basePath: tilesetJson.basePath,
+    onTraversalEnd: traversalEnd
+  });
+
+  const viewport = new WebMercatorViewport({
+    altitude: 1.5,
+    bearing: 0,
+    far: 1000,
+    fovy: 50,
+    height: 600,
+    id: 'view0',
+    latitude: 40.049483884253355,
+    longitude: -75.60783109310839,
+    maxPitch: 85,
+    maxZoom: 30,
+    minPitch: 0,
+    minZoom: 2,
+    modelMatrix: null,
+    near: 0.1,
+    pitch: 45,
+    projectionMatrix: null,
+    width: 1848,
+    zoom: 12.660812211760435
+  });
+  traverser.traverse(tileset.root, getFrameState(viewport, 0), {});
+  function traversalEnd() {
+    t.ok(traverser);
```

Comment:
Great. Any chance we could make an assertion on which tiles were selected? A separate PR is fine.

Reply:
Yes. It should be test cases for sub-classes - 3DTiles and I3S traversers

Reply:
OK. I will just say that this is a problem with the current setup. The various loaders are supposed to normalize data and there should only be one traversal algorithm, not one per format. This is a bigger effort but we should tackle it at some point instead of patching on the current hacky solution. Perhaps we can find smaller steps to gradually move us to a better place.